apps/prairielearn/src/sync/infofile.ts

```diff
@@ -98,12 +97,7 @@ export function addWarnings(infoFile, warnings) {
   infoFile.warnings = infoFile.warnings.concat(warnings);
 }
 
-/**
- * @template T
- * @param {Partial<Pick<InfoFile<T>, 'uuid' | 'data'>>} infoFile
- * @returns {InfoFile<T>}
- */
-export function makeInfoFile(infoFile = {}) {
+export function makeInfoFile<T>(infoFile: Omit<InfoFile<T>, 'errors' | 'warnings'> = {}) {
```

Comment:
In this file I only converted the strictly necessary to make it typecheck. Remaining functions have not been changed yet, since changing them would cause Git to lose the renaming of the file for history.